src/components/CollectionProvider/CollectionProvider.tsx

```diff
@@ -1,28 +1,49 @@
 import * as React from 'react'
-import { Props } from './CollectionProvider.types'
+import equal from 'fast-deep-equal'
+import { DEFAULT_ITEMS_PAGE_SIZE, DEFAULT_ITEMS_PAGE, Props } from './CollectionProvider.types'
 
 export default class CollectionProvider extends React.PureComponent<Props> {
+  fetchCollectionItems(itemsPage: number | number[] = DEFAULT_ITEMS_PAGE) {
+    const { id, onFetchCollectionItems, itemsPageSize } = this.props
+    if (id) {
+      const pageSize = itemsPageSize || DEFAULT_ITEMS_PAGE_SIZE
+      onFetchCollectionItems(id, itemsPage, pageSize)
```

Comment:
I had to introduced this logic when doing the `fetch all pages` feature, from the TP detail page. In that case, I needed ALL the items (`items`) for the collection (to set all the ids as selected in the component's state) and also the paginated ones (because the view it's still being paginated). An alternative to that could be to have a selector in the TP detail page to grab all the collection items and keep the `CollectionProvider` returning just the paginated ones, but I'm not sure it's somehow different to this.